src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1294,6 +1298,20 @@ private bool ShouldAsyncBeTreatedAsModifier(bool parsingStatementNotDeclaration)
                 return true;
             }
 
+            if (!alreadyHadAsyncModifier &&
```

Comment:
This seems extremely marginal. It only affects things when typing the last member of a type, and only if that last manager is followed by extremely specific cases. This feels like over fitting a solution to solve exactly one case.

Reply:
Yes. When investigating the issue, I was a bit surprised, but in almost all cases, when typing a member before other one, `async` is recognized as a modifier. The issue is only about the last member. However, It doesn't look like over fitting fo me, just an edge case

Reply:
That's seems very strange to me. Even just following with an attribute would give `async[` which I would expect was parsed as an array type.

Reply:
i'm also surprised by this cahnge given the code below.  it looks like we already look for this on line: https://github.com/dotnet/roslyn/pull/60403/files#diff-a655d393d9799eabace5e5109383232749be8052f1fd825fc52b9f170908e838L1350

Reply:
ah, n/m  that's if we have `async type-name`.